src/mark.c

```diff
@@ -170,13 +182,23 @@ gboolean r_mark_good(RaucSlot *slot, GError **error)
 	g_return_val_if_fail(slot, FALSE);
 	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
 
+	/* It is more sensible to mark the slot as good before locking the counter. As in case of a power fail between
+	 * marking and locking, we'd have a system with a working fallback to the old stot, so in case of problems
```

Comment:
```suggestion
	 * marking and locking, we'd have a system with a working fallback to the old slot, so in case of problems
```

I suggest pointing out that on next boot, the mark good still has a chance to lock, so it's eventually consistent of sorts. Also I think comment style used differs from the usual RAUC convention. Please align.